validator/gke_restrict_pod_traffic.rego

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GKERestrictPodTrafficConstraintV1
+
+import data.validator.gcp.lib as lib
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	asset := input.asset
+	asset.asset_type == "container.googleapis.com/Cluster"
+
+	container := asset.resource.data
+	not check_all_enabled(container)
+
+	message := sprintf("%v doesn't restrict traffic among pods with a network policy.", [asset.name])
+	metadata := {"resource": asset.name}
+}
+
+###########################
+# Rule Utilities
+###########################
+check_all_enabled(container) {
+	network_policy_config_enabled(container) == true
+	pod_security_policy_config_enabled(container) == true
+	network_policy_enabled(container) == true
+}
+
+network_policy_config_enabled(container) {
+	addons_config := lib.get_default(container, "addonsConfig", {})
+	networkPolicyConfig := lib.get_default(addons_config, "networkPolicyConfig", {})
+	network_policy_config_disabled := lib.get_default(networkPolicyConfig, "disabled", false)
```

Comment:
not knowing anything about network policy config, it's enabled by default and needs to be disabled explicitly?
Can we include a pointer to the documentation?